### PR TITLE
[BEAM-1579] Add a Batch-Surgery API

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---
This takes a list, applies all overrides in the order of the list, and
validates that no more overrides are applicable.

Port the DirectRunner to this API

### CODE REVIEW

**Comment by tgroh** [comment]
R: @jkff

**Comment by tgroh** [comment]
retest this please

File: sdks/java/core/src/main/java/org/apache/beam/sdk/Pipeline.java [hunk 1]
```diff
@@ -172,14 +175,67 @@ public PBegin begin() {
...
   }
 
+  /**
+   * Replaces all nodes that match a {@link PTransformOverride} in this pipeline. Overrides are
+   * applied in the order they are present within the list.
+   *
+   * <p>After all nodes are replaced, ensures that no nodes in the updated graph match any of the
+   * overrides.
+   */
+  public void replaceAll(List<PTransformOverride> overrides) {
```
Comment (jkff) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/Pipeline.java]: Add a test for this behavior?

Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/Pipeline.java]: Done.

**Comment by jkff** [comment]
LGTM, merging.

**Pull request closed without merging**